core/src/main/scala/org/apache/spark/util/collection/ExternalAppendOnlyMap.scala

```diff
@@ -267,7 +273,7 @@ class ExternalAppendOnlyMap[K, V, C](
    */
   def destructiveIterator(inMemoryIterator: Iterator[(K, C)]): Iterator[(K, C)] = {
     readingIterator = new SpillableIterator(inMemoryIterator)
-    readingIterator
+    readingIterator.toCompletionIterator
```

Comment:
This change the original behavior of `destructiveIterator `. I'd prefer do like this:

```
CompletionIterator[(K, C), Iterator[(K, C)]](
destructiveIterator(currentMap.iterator), readingIterator.destroy)
```
which keep compatibility with current code, and do not introduce unnecessary function.

Reply:
What behavior does it change? Your suggested codes does exactly the same but is less streamlined and relies on an intermediate value (fortunately it's already a member variable)

Reply:
`destructiveIterator` should  just return a destructive iterator (especially for map buffer) as it's function name implies, and it it none business of `CompletionIterator `. And developers should be free to define the complete function for the returned destructive iterator, in case of we want a different one somewhere else in future.

 > Your suggested codes does exactly the same but is less streamlined 

I don't think this little change will pay a huge influence on `streamlined `.

> and relies on an intermediate value (fortunately it's already a member variable)

The current fix leads to this, not me. And even this variable is not a member variable, we can define a temp local variable. It's not a big deal.